nnfabrik/main.py

```
@@ -351,3 +351,77 @@ class Seed(dj.Manual):
+        use_common_fabrikant (bool, optional): If True, new tables will depend on the
+           common Fabrikant table. If False, new copy of Fabrikant will be created and used. 
+           Defaults to True.
+        use_common_seed (bool, optional): If True, new tables will depend on the
+           common Seed table. If False, new copy of Seed will be created and used. 
+           Defaults to False.
+        module_name (str, optional): Name property of the returned Python module object.
+            Defaults to None, in which case the name of the schema will be used.
+        context (dict, optional): If non None value is provided, then a module is not created and
+            instead the tables are defined inside the context.
+        spawn_existing_tables (bool, optional): If True, perform `spawn_missing_tables` operation
+            onto the newly created table. Defaults to False.
+
+    Returns:
+        Python Module object or None: If `context` was None, a new Python module containing 
+            nnfabrik tables defined under the schema. The module's schema property points 
+            to the schema object as well. Otherwise, nothing is returned.
+    """
+    if isinstance(schema, str):
+        schema = CustomSchema(schema)
+
+    tables = [Fabrikant, Model, Dataset, Trainer]
+
+    if not use_common_seed:
+        tables.append(Seed)
+
+    module = None
+    if context is None:
+        module_name = schema.database if module_name is None else module_name
+        module = types.ModuleType(module_name)
+        context = module.__dict__
+
+    context["schema"] = schema
+
+    # spawn all existing tables into the module
+    # TODO: replace with a cheaper check operation
+    temp_context = context if spawn_existing_tables else {}
+    schema.spawn_missing_classes(temp_context)
+
+    if use_common_fabrikant:
+        if "Fabrikant" in temp_context:
+            warnings.warn(
```

Raise an exception instead of warning

---

Addressed